src/track/macro.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <QtCore>
+#include <memory>
+
+#include "engine/engine.h"
+#include "proto/macro.pb.h"
+#include "track/macroaction.h"
+#include "util/db/dbid.h"
+namespace proto = mixxx::track::io;
+
+const QLoggingCategory macroLoggingCategory("macros");
+constexpr int kMacrosPerChannel = 16;
+
+/// A Macro stores a list of MacroActions as well as its current state and label.
+class Macro {
+  public:
+    static QByteArray serialize(const QList<MacroAction>& actions);
+    static QList<MacroAction> deserialize(const QByteArray& serialized);
+
+    enum class StateFlag {
+        Enabled = 1u,
+        Looped = 2u,
+    };
+    Q_DECLARE_FLAGS(State, StateFlag);
+
+    explicit Macro(const QList<MacroAction>& actions = {},
+            const QString& label = "",
```

Comment:
Prefer to use a null string instead of an empty string.

Are all those default arguments really needed? This allows to create a macro in 5 different ways. Prefer to expose a limited API.

Example: A DbId must never be provided explicitly except by the DAO.